From b495437d4fad999828139ca86f8ecc27573eb859 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Thu, 23 Apr 2020 15:21:02 -0700
Subject: [PATCH] Fix compatibility with curl-7.62.0 and newer

* fails with newer curl
  CURLE_SSL_CACERT was replaced with CURLE_PEER_FAILED_VERIFICATION
  https://github.com/curl/curl/pull/3291/files

ros1-melodic-dunfell/tmp-glibc/work/aarch64-oe-linux/rc-hand-eye-calibration-client/2.7.0-1-r0/git/opt/cpr/cpr/error.cpp: In static member function 'static cpr::ErrorCode cpr::Error::getErrorCodeForCurlError(int32_t)':
ros1-melodic-dunfell/tmp-glibc/work/aarch64-oe-linux/rc-hand-eye-calibration-client/2.7.0-1-r0/git/opt/cpr/cpr/error.cpp:41:9: error: duplicate case value
   41 |         case CURLE_SSL_CACERT:
      |         ^~~~
ros1-melodic-dunfell/tmp-glibc/work/aarch64-oe-linux/rc-hand-eye-calibration-client/2.7.0-1-r0/git/opt/cpr/cpr/error.cpp:25:9: note: previously used here
   25 |         case CURLE_PEER_FAILED_VERIFICATION:
      |         ^~~~

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 opt/cpr/cpr/error.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/opt/cpr/cpr/error.cpp b/opt/cpr/cpr/error.cpp
index 713cb10c..4143f93b 100644
--- a/opt/cpr/cpr/error.cpp
+++ b/opt/cpr/cpr/error.cpp
@@ -38,8 +38,6 @@ ErrorCode Error::getErrorCodeForCurlError(std::int32_t curl_code) {
             return ErrorCode::SSL_LOCAL_CERTIFICATE_ERROR;
         case CURLE_SSL_CIPHER:
             return ErrorCode::GENERIC_SSL_ERROR;
-        case CURLE_SSL_CACERT:
-            return ErrorCode::SSL_CACERT_ERROR;
         case CURLE_USE_SSL_FAILED:
             return ErrorCode::GENERIC_SSL_ERROR;
         case CURLE_SSL_ENGINE_INITFAILED:
